TESTING.asciidoc

```diff
@@ -142,6 +142,47 @@ Now the http.p12 file will be placed in the config directory of the running clus
 Assuming you have the http.ssl.keystore setup correctly, you can now use HTTPS with ./gradlew run without the risk
 of accidentally committing your local configurations.
 
+==== Multiple nodes in the test cluster for ./gradlew run
+
+Another desired customization for ./gradlew run might be to run multiple
+nodes with different setting for each node. For example, you may want to debug a coordinating only node that fans out
+to one or more data nodes. To do this, increase the numberOfNodes attribute for the test cluster configuration.
+-------------------------------------
+testClusters.register("runTask") {
+      ...
+      numberOfNodes = 2
+    }
+}
+-------------------------------------
+
+Then customize the run task itself to create different configurations per node. For example, the following will
+instruct the first node (:9200) to be a coordinating only node, and all other nodes to be master, data_hot, data_content
+nodes. Since we are changing the default master node configuration we must also update the default initial_master_nodes.
+Finally, we need to "restart" the cluster to pick up the changes. However, it does not actually start/stop/start
+the cluster since we have not started the cluster yet and this instructs the plugin to update the configuration.
+
+-------------------------------------
+tasks.register("run", RunTask) {
+  useCluster testClusters.named("runTask")
+  description = 'Runs elasticsearch in the foreground'
+  group = 'Verification'
+
+  doFirst {
```

Comment:
Why would we do this here rather than just configure the test cluster this way to begin with up above?

Reply:
If i understand correctly the testCluster is not realized yet  (i.e. only a provider and only registered) and wasn't sure how to get a reference to each of the nodes in the above task. By placing it here and I am realizing the cluster in this task and re-configuring it a bit before but before it is started.

Reply:
You would configure it in exactly the same way you set the number of nodes. Any and all configuration can be done there, it's simply deferred until the cluster is actually required.

Reply:
maybe I am not understanding but putting this config as-is just below numberOfNodes in the above task registration results in `Could not create domain object 'runTask' (ElasticsearchCluster)` https://gradle-enterprise.elastic.co/s/xjmtjfgceprli similar results wrapped in a doLast. 

I did find some other examples and the following would work inside the testCluster.register("runTask")
```
      nodes."runTask-0".setting('node.roles', '[]')
      nodes."runTask-1".setting('node.roles', '[master,data_hot,data_content]')
      nodes."runTask-0".setting('cluster.initial_master_nodes', 'runTask-1')
      nodes."runTask-1".setting('cluster.initial_master_nodes', 'runTask-1')
```
I do slightly prefer the example in the PR, especially one i redo the run-ccs to work like that.

Reply:
For context [here is my WIP](https://github.com/elastic/elasticsearch/compare/main...jakelandis%3Are_do_run_ccs?body=&expand=1) for re-doing run-css.

Reply:
That error is because you are trying to call `restart()` on a cluster that hasn't started. Since we are configuring everything before the cluster has started, there's no need for a restart at all in this scenario so you should just be able to remote that line.

Reply:
Thanks !  I like that much better. I updated the doc via [`9b031d4` (#89563)](https://github.com/elastic/elasticsearch/pull/89563/commits/9b031d42536eb624380f28315e282ccb9ae12960)